sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1085,33 +1107,72 @@ private static void throwForTypeMismatches(List<TypeMismatch> mismatches) {
     }
     throwForGettersWithInconsistentJsonIgnore(incompletelyIgnoredGetters);
     throwForSettersWithJsonIgnore(ignoredSetters);
+  }
 
+  /**
+   * Validates that every bean property of any interface derived from {@link PipelineOptions}
+   * must have a getter and setter method.
+   *
+   * @param iface The interface to validate.
+   * @param descriptors The list of {@link PropertyDescriptor}s representing all valid bean
+   * properties of {@code iface}.
+   */
+  private static void validateGettersSetters(
+      Class<? extends PipelineOptions> iface,
+      List<PropertyDescriptor> descriptors) {
     List<MissingBeanMethod> missingBeanMethods = new ArrayList<>();
-    // Verify that each property has a matching read and write method.
     for (PropertyDescriptor propertyDescriptor : descriptors) {
       if (!(IGNORED_METHODS.contains(propertyDescriptor.getWriteMethod())
-        || propertyDescriptor.getReadMethod() != null)) {
+          || propertyDescriptor.getReadMethod() != null)) {
         MissingBeanMethod method = new MissingBeanMethod();
         method.property = propertyDescriptor;
         method.methodType = "getter";
         missingBeanMethods.add(method);
         continue;
       }
       if (!(IGNORED_METHODS.contains(propertyDescriptor.getReadMethod())
-              || propertyDescriptor.getWriteMethod() != null)) {
+          || propertyDescriptor.getWriteMethod() != null)) {
         MissingBeanMethod method = new MissingBeanMethod();
         method.property = propertyDescriptor;
         method.methodType = "setter";
         missingBeanMethods.add(method);
         continue;
       }
-      methods.add(propertyDescriptor.getReadMethod());
-      methods.add(propertyDescriptor.getWriteMethod());
```

Comment:
It seems like it wouldn't NPE because throwForMissingBeanMethod should have thrown since just above we add methods to missingBeanMethods if we don't have both the setter and getter.